packages/dds/merge-tree/src/endOfTreeSegment.ts

```diff
@@ -24,18 +25,18 @@ import { depthFirstNodeWalk, NodeAction } from "./mergeTreeNodeWalk";
  */
 export class EndOfTreeSegment implements ISegment, IRemovalInfo {
     type: string = "EndOfTreeSegment";
-    private readonly root: IMergeBlock;
+    private readonly mergeTree: MergeTree;
     constructor(segmentOrNode: IMergeNode) {
-        let maybeRoot: IMergeBlock | undefined = segmentOrNode.isLeaf()
-            ? segmentOrNode.parent
-            : segmentOrNode;
-        while (maybeRoot?.parent !== undefined) {
+        let maybeRoot: IRootMergeBlock | undefined = segmentOrNode.isLeaf()
+        ? segmentOrNode.parent
```

Comment:
i'd revert whitespace changes here and on the `while` line

(soon we should have prettier and won't need to worry about this kind of thing.. :))